src/wkt/src/message.rs

```diff
@@ -20,27 +20,52 @@ use crate::AnyError as Error;
 /// A trait that must be implemented by all messages.
 ///
 /// Messages sent to and received from Google Cloud services may be wrapped in
-/// [Any][crate::any::Any]. `Any` uses a `@type` field to encoding the type
+/// [Any][crate::any::Any]. `Any` uses a `@type` field to encode the type
 /// name and then validates extraction and insertion against this type.
 pub trait Message {
     /// The typename of this message.
     fn typename() -> &'static str;
 
+    /// Returns the serializer for this message type
     #[doc(hidden)]
-    /// Store the value into a JSON object.
-    fn to_map(&self) -> Result<Map, Error>
+    fn serializer() -> Box<dyn MessageSerializer<Self>>
```

Comment:
We should prefer static dispatch. It is more performant. We only use dynamic dispatch when needed.

For any given `T`, there is only ever one `MessageSerializer` returned, so static dispatch will work just fine.

Reply:
Reverted back to static dispatch.